homeassistant/components/roku.py

```diff
@@ -0,0 +1,115 @@
+"""
+Support for Roku platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/roku/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.discovery import SERVICE_ROKU
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-roku==3.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'roku'
+
+SERVICE_SCAN = 'roku_scan'
+
+ATTR_ROKU = 'roku'
+
+DATA_ROKU = 'data_roku'
+
+NOTIFICATION_ID = 'roku_notification'
+NOTIFICATION_TITLE = 'Roku Setup'
+NOTIFICATION_SCAN_ID = 'roku_scan_notification'
+NOTIFICATION_SCAN_TITLE = 'Roku Scan'
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [vol.Schema({
+        vol.Required(CONF_HOST): cv.string
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+# Currently no attributes but it might change later
+ROKU_SCAN_SCHEMA = vol.Schema({})
+
+
+def setup(hass, config):
+    """Set up the Roku component."""
+    hass.data[DATA_ROKU] = {}
+
+    def service_handler(service):
+        """Handle service calls."""
+        if service.service == SERVICE_SCAN:
+            scan_for_rokus(hass)
+
+    def roku_discovered(service, info):
+        """Set up an Roku that was auto discovered."""
+        _setup_roku(hass, config, {
+            CONF_HOST: info['host']
+        })
+
+    discovery.listen(hass, SERVICE_ROKU, roku_discovered)
+
+    for conf in config.get(DOMAIN, []):
```

Comment:
Use `dict[key]`.

Reply:
As I said before, this platform is auto-detect and doesn't need `roku:` in the configuration to activate this.

Reply:
Right! I missed this, sorry.